### DISCUSSION TITLE
Easiest way to copy schema of existing table to create a new table?

### DISCUSSION THREAD

**Discussion by osseonews** [nhost/nhost#3068] [Q&A] [answered]
Sorry, if this seems basic, and i've used all the tools in the past, like pgadmin, but it's all a big pain. So my basic question is, using Nhost, what's the easiest way to simply copy the exact schema of a table and duplicate for a new table? Do you have any existing scripts to do this? Our use case is simple. Very often, we create a table for dev and then once we have tested it, we want to just duplicate the exact schema for a live implementation, without having to redo everything in a manual process. BTW, we create our table with the Hasura dashboard, as it's just the easiest way to do it. If we can use low code, we will. Thanks.

**Comment by dbarrosop** [comment]
The easiest way would to be use the "local development" workflow with multiple connected projects to the same repo but different branches (i.e. a dev branch and a production branch):

https://docs.nhost.io/guides/cli/local-development

**Reply by osseonews** [reply]
Thanks. I was looking online at Postgre documentation and would something like this work if we ran it in the dashboard for SQL? `CREATE TABLE new_table (LIKE existing_table INCLUDING ALL);
 `
or something like this: ```
create table my_table_copy as
  select * from my_table
with no data
```

**Reply by osseonews** [reply, marked as answer]
So, just experimented a bit and the command below works fine, in case anyone needs this. It just doesn't create any primary keys, constraints, indexes etc. You need to add those after, but that's easy via the ALTER table command. 

```
CREATE TABLE new_table AS 
TABLE old_table 
WITH NO DATA;
```

**Reply by dbarrosop** [reply]
Glad you figure it out. For anyone reading this though, we don't recommend using the same project for dev and production as any mistake could incur a degradation of service or even data loss. A much better approach would be to have different projects for each environment as needed (prod, dev, staging, etc)

**Reply by osseonews** [reply]
Hey, I agree, but that gets expensive. Maybe you can have a sort of "dev" pricing that sits between, starter and pro. $5 a month for a really basic set up that can be used for dev (the starter isn't useful because it sleeps). I'd sign up for that. Incremental revenue for you guys.

**Reply by dbarrosop** [reply]
I think you may have missed our latest announcement regarding organizations, please, check it out; https://nhost.io/blog/organization-billing

**Reply by osseonews** [reply]
You are right, I  missed that and was wondering what Workspace Legacy meant in the dashboard. Is there any way someone can help us transition to the new organization? honestly, don't have a clue what to do.
